### PR TITLE
IBX-771: Implemented new design for ckeditor

### PR DESCRIPTION
**Pull request by dew326**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | https://issues.ibexa.co/browse/IBX-771
| **Requires**      | https://github.com/ezsystems/ezplatform-admin-ui/pull/1902
| **New feature**    | yes
| **Target version** | 4.0
| **BC breaks**      | no
| **Tests pass**     | yes
| **Doc needed**     | no

**TODO**:
- [x] Implement feature / fix a bug.
- [x] Ask for Code Review.

### CODE REVIEW

File: src/bundle/Resources/public/js/CKEditor/custom-tags/block-custom-tag/custom-tag-editing.js [hunk 1]
```diff
@@ -31,11 +31,35 @@ class IbexaCustomTagEditing extends Plugin {
...
             view: (modelElement, { writer: downcastWriter }) => {
+                const customTagName = modelElement.getAttribute('customTagName');
                 const container = downcastWriter.createContainerElement('div', {
                     'data-ezelement': 'eztemplate',
-                    'data-ezname': modelElement.getAttribute('customTagName'),
-                    class: 'ez-custom-tag',
+                    'data-ezname': customTagName,
+                    class: 'ibexa-custom-tag',
                 });
+                const header = downcastWriter.createUIElement('div', { class: 'ibexa-custom-tag__header' }, function(domDocument) {
```
Comment (lucasOsti) [inline src/bundle/Resources/public/js/CKEditor/custom-tags/block-custom-tag/custom-tag-editing.js:40]: why not arrow function?

File: src/bundle/Resources/public/js/CKEditor/link/ui/link-form-view.js [hunk 2]
```diff
@@ -16,16 +16,56 @@ class IbexaLinkFormView extends View {
         this.selectContentButtonView = this.createButton('Select content', null, 'ibexa-btn--select-content');
         this.urlInputView = this.createTextInput('Link to');
         this.titleView = this.createTextInput('Title');
-        this.targetSwitcherView = this.createBoolean('Open in');
+        this.targetSwitcherView = this.createBoolean('Open in tab');
```
Comment (lucasOsti) [inline src/bundle/Resources/public/js/CKEditor/link/ui/link-form-view.js:19]: use traslation

File: src/bundle/Resources/public/scss/_general.scss [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+            }
+
+            &.ck-widget_with-selection-handle.ck-widget_selected .ck-widget__selection-handle,
+            &.ck-widget_with-selection-handle.ck-widget_selected:hover .ck-widget__selection-handle,
+            &.ck-widget_selected > .ck-widget__type-around > .ck-widget__type-around__button {
+                background: $ibexa-color-dark;
+            }
+
+            &.table,
+            table {
```
Comment (lucasOsti) [inline src/bundle/Resources/public/scss/_general.scss:46]: I think we should avoid styling on tags

File: src/bundle/Resources/public/js/CKEditor/custom-tags/block-custom-tag/custom-tag-editing.js (hunk 1 above)
Comment (dew326) [reply src/bundle/Resources/public/js/CKEditor/custom-tags/block-custom-tag/custom-tag-editing.js:40]: With arrow function this is lost and CKEditor throws an error.

File: src/bundle/Resources/public/js/CKEditor/link/ui/link-form-view.js (hunk 2 above)
Comment (dew326) [reply src/bundle/Resources/public/js/CKEditor/link/ui/link-form-view.js:19]: Translations will be added later

File: src/bundle/Resources/public/scss/_general.scss (hunk 3 above)
Comment (dew326) [reply src/bundle/Resources/public/scss/_general.scss:46]: We should but this overrides CKEditor styles

File: src/bundle/Resources/public/js/CKEditor/custom-attributes/custom-attributes-ui.js [hunk 4]
```diff
@@ -67,7 +67,8 @@ class IbexaAttributesUI extends Plugin {
 
             return output;
         }, {});
-        let classesValue = areCustomAttributesSet ? parentElement.getAttribute('custom-classes') : customClasses.defaultValue;
+        const defaultCustomClasses = customClasses ? customClasses.defaultValue : '';
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/js/CKEditor/custom-attributes/custom-attributes-ui.js:70]:
```suggestion
        const defaultCustomClasses = customClasses?.defaultValue ?? '';
```
;P

File: src/bundle/Resources/public/js/CKEditor/custom-attributes/ui/custom-attributes-form-view.js [hunk 5]
```diff
@@ -41,11 +41,41 @@ class IbexaCustomAttributesFormView extends View {
...
+                class: 'ibexa-ckeditor-balloon-form',
             },
-            children: this.children,
+            children: [
+                {
+                    tag: 'div',
+                    attributes: {
+                        class: 'ibexa-ckeditor-balloon-form__header',
+                    },
+                    children: ['Custom Attributes'],
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/js/CKEditor/custom-attributes/ui/custom-attributes-form-view.js:54]: No translation?

File: src/bundle/Resources/public/js/CKEditor/custom-tags/block-custom-tag/custom-tag-editing.js [hunk 6]
```diff
@@ -31,11 +31,35 @@ class IbexaCustomTagEditing extends Plugin {
...
+                    const domElement = this.toDomElement(domDocument);
+
+                    domElement.innerHTML = `
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/js/CKEditor/custom-tags/block-custom-tag/custom-tag-editing.js:43]: Cannot you put it in twig and pass as `data-template`?

File: src/bundle/Resources/public/scss/_balloon-form.scss [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+
+                        &__inner {
+                            border-radius: 50%;
+                            width: calculateRem(10px);
+                            height: calculateRem(10px);
+                            transform: translate(calculateRem(2px), calculateRem(2px));
+                        }
+
+                        &:hover {
+                            background: red;
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/scss/_balloon-form.scss:55]: red?

**Review by GrabowskiM — CHANGES_REQUESTED** [review]

File: src/bundle/Resources/public/js/CKEditor/custom-tags/block-custom-tag/custom-tag-editing.js (hunk 6 above)
Comment (dew326) [reply src/bundle/Resources/public/js/CKEditor/custom-tags/block-custom-tag/custom-tag-editing.js:43]: Those should be something outside of the twig templates.

File: src/bundle/Resources/public/js/CKEditor/custom-attributes/ui/custom-attributes-form-view.js (hunk 5 above)
Comment (dew326) [reply src/bundle/Resources/public/js/CKEditor/custom-attributes/ui/custom-attributes-form-view.js:54]: Will be added later in bulk

File: src/bundle/Resources/public/js/CKEditor/embed/image/embed-image-toolbar.js [hunk 8]
```diff
@@ -8,7 +8,7 @@ class IbexaEmbedImageToolbar extends Plugin {
 
     getSelectedEmbedImageWidget(selection) {
         const viewElement = selection.getSelectedElement();
-        const isEmbedImage = viewElement && viewElement.hasClass('ez-embed-type-image');
+        const isEmbedImage = viewElement && viewElement.hasClass('ibexa-embed-type-image');
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/js/CKEditor/embed/image/embed-image-toolbar.js:11]:
```suggestion
        const isEmbedImage = viewElement?.hasClass('ibexa-embed-type-image');
```
prophet of optional chaining has come...

File: src/bundle/Resources/public/js/CKEditor/link/ui/link-form-view.js [hunk 9]
```diff
@@ -16,16 +16,56 @@ class IbexaLinkFormView extends View {
...
+                            children: [
+                                this.children.first,
+                                {
+                                    tag: 'div',
+                                    attributes: {
+                                        class: 'ibexa-ckeditor-balloon-form__separator',
+                                    },
+                                    children: ['Or'],
+                                },
+                                ...this.children.filter((child, index) => index !== 0),
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/js/CKEditor/link/ui/link-form-view.js:56]:
```suggestion
                                ...this.children.slice(1),
```
Wouldn't be better and faster?

**Review by GrabowskiM — CHANGES_REQUESTED** [review]

File: src/bundle/Resources/public/js/CKEditor/link/ui/link-form-view.js (hunk 9 above)
Comment (dew326) [reply src/bundle/Resources/public/js/CKEditor/link/ui/link-form-view.js:56]: Would be if this would be an array ;)
This is ckeditor collection.

**Review by lucasOsti — APPROVED** [review]

**Review by tischsoic — APPROVED** [review]

**Pull request merged**